### ISSUE TITLE
Feature Request- Show IV of pokemon

### ISSUE THREAD

**Issue by 2Awesome2** [AHAAAAAAA/PokemonGo-Map#1930]
All pokemon that spawn have the same IV just different CP values based on the trainer level. Is it possible for the map to show the IV (and better yet, filter based upon IVs)?

**Comment by stereoa** [comment]
I could be wrong and probably shouldn't be commenting, but isn't that information only sent after catch? The map client would have to catch all the pokemon it encounters then.

**Comment by medghaim** [comment]
This would require people to use their own accounts, which we do not encourage. CP and I'd assume IV's as well, are tied to your account. A 54 Dragonite to me, could be a 1542 dragonite to you

**Comment by 2Awesome2** [comment]
CP of pokemon is based upon the seed of the trainers level (all level 20 trainers would get the same CP pokemon)

IV is the same for everyone

**Comment by xrobau** [comment]
A simpler explanation is that the IV references how far along the 'curve' the pokemon is.

**Comment by QuadTog** [comment]
Unless it seeds CP randomly for each trainer level I cant see how that works.

While I do have evidence of 3 lvl21 trainers catching the exact same CP Pokemon repeatedly. I also have evidence of low level trainers catching significantly higher level CP Pokemon then me.

IV's indicate how strong the Pokemon is from its baseline stats; it can be anywhere from +0-15 for ATK, DEF and STAM. Since CP is calculated off the Pokemon's stats, higher IV's can equate up to a +-500CP difference for Pokemon.

I still can't see how you're going to calculate the overall IV's of a Pokemon without first catching it to view it's level up cost.

**Comment by 2Awesome2** [comment]
CP=f(trainer level), [and it has to be below the max CP you can get for that level) yes it's possible for a lower level to get a higher CP than some one with a higher trainer level.

What data is able to be viewed besides time remaining and which pokemon it is? (without catching it)

Is it just not able to be seen without catching it?

**Comment by BillyWM** [comment]
>IV is the same for everyone

This seems speculative.

>While I do have evidence of 3 lvl21 trainers catching the exact same CP Pokemon repeatedly.

Have they all run the IV calculation on what they caught? Maybe /r/thesilphroad would do this research if it hasn't already?

**Comment by 2Awesome2** [comment]
Yes IVs were compared between people with different trainer levels getting the same IV

**Comment by kyleshrader** [comment]
I was under the impression IV/CP are not determinable pre-catch. At least, no current APIs support viewing this Pre-catch. also CP is a function of level so we would want to add some client side level field if this were possible.

**Comment by Talamar1** [comment]
It is available without a catch, but might require at least attempting a catch.  Here is output from a farming bot:
```
[21:36:33] (PKMN) (CatchEscape) | (Normal) Weedle Lvl: 15 CP: (146/192) IV: 27.00% | Chance: 38.66% | 29.35m dist | with a PokeBall (48 left). |
[21:36:36] (PKMN) (CatchEscape Attempt #2) | (Normal) Weedle Lvl: 15 CP: (146/192) IV: 27.00% | Chance: 38.66% | 29.35m dist | with a PokeBall (47 left). |
[21:36:39] (PKMN) (CatchSuccess Attempt #3) | (Normal) Weedle Lvl: 15 CP: (146/192) IV: 27.00% | Chance: 38.66% | 29.35m dist | with a PokeBall (46 left). | Candies: 6
```
As you can see, it shows the IV for the Weedle before being caught (2 fails).

The IV & level is always the same, determined at spawn.  The HP/CP vary based on the trainer (you can test this by using a Snipe program and have 3-5 people at different levels snipe the same pokemon).
